### ISSUE TITLE
Logging in to Pandora Bar

### ISSUE THREAD

**Issue by n8klayko** [david4shure/Pianobar-Web#2]
As far as I can tell, I have installed everything properly. I can get to my pi running Pianobar and Pianobar-Web by going to 192.168.0.130:8080. However, I am prompted for a email and password. I entered my Pandora email and password and I get the error "invalid login". Any Ideas?

**Comment by david4shure** [comment]
What version of pianobar are you running? and can you log into piano bar on the command line with those credentials?

**Comment by david4shure** [comment]
Also could you give me a bit more information regarding your environment?

**Comment by n8klayko** [comment]
I compiled the most recent version of pianobar and the credentials work
fine in just pianobar. I grabbed a second SD card to try from a fresh
install of raspbian.

**Comment by n8klayko** [comment]
I was running the most recent version of raspbian

**Comment by n8klayko** [comment]
I got everything working with a fresh install.

On Thu, Sep 24, 2015 at 9:24 PM, Nathan Klayko nathanklayko@gmail.com
wrote:

> I was running the most recent version of raspbian
> On Sep 24, 2015 8:41 PM, "Nathan Klayko" nathanklayko@gmail.com wrote:
> 
> > I compiled the most recent version of pianobar and the credentials work
> > fine in just pianobar. I grabbed a second SD card to try from a fresh
> > install of raspbian.
> > On Sep 24, 2015 8:23 PM, "David Shure" notifications@github.com wrote:
> > 
> > > What version of pianobar are you running? and can you log into piano bar
> > > on the command line with those credentials?
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/david4shure/Pianobar-Web/issues/2#issuecomment-143088080
> > > .

## 

_Nathan Klayko_
Technology Consultant & Contractor
The Summit Radio Station
JN Audio & Tech
Email: nathanklayko@gmail.com
Phone: (330) 620-7574

**Comment by david4shure** [comment]
Glad to hear that, if you encounter any additional issues, or want any new features, just open an issue and I'll see what I can do.

**Comment by n8klayko** [comment]
David, I was wondering if you had ever made Pandora Bar able to add and
remove channels?

On Sat, Sep 26, 2015 at 1:46 PM, David Shure notifications@github.com
wrote:

> Glad to hear that, if you encounter any additional issues, or want any new
> features, just open an issue and I'll see what I can do.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/david4shure/Pianobar-Web/issues/2#issuecomment-143475293
> .

## 

_Nathan Klayko_
Technology Consultant & Contractor
The Summit Radio Station
JN Audio & Tech
Email: nathanklayko@gmail.com
Phone: (330) 620-7574

**Comment by david4shure** [comment]
@n8klayko No, I have not implemented that feature since I haven't touched this in awhile, so feel free to open an issue, and I'll get around to it, or you could also open a pull request yourself.

**Comment by david4shure** [comment]
for the record 

https://github.com/david4shure/Pianobar-Web/issues/3

**Comment by jbobspants** [comment]
I'm having the same issue, just get invalid logon when I input my Pandora login info in the web interface.

Running on Volumio 1.55 (on Raspberry Pi). Latest version of Pianobar compiled from source, it runs great running pianobar directly from the command line, but Pianobar-Web does not want to login.

I originally had my email and password statically configured in ~/.config/pianobar/config so pianobar would log in automatically. I commented out those lines so now pianobar prompts for logins, I can still login manually with pianobar but not the web interface.

Here's the console output when I try to login:

pi@volumio:~/builds/Pianobar-Web$ python pianobar_web.py 
Bottle v0.13-dev server starting up (using WSGIRefServer())...
Listening on http://0.0.0.0:8080/
Hit Ctrl-C to quit.

192.168.1.246 - - [01/Feb/2016 14:26:14] "GET / HTTP/1.1" 303 0
192.168.1.246 - - [01/Feb/2016 14:26:14] "GET /login HTTP/1.1" 200 967
192.168.1.246 - - [01/Feb/2016 14:26:28] "GET /static/foundation.min.css HTTP/1.1" 304 0
192.168.1.246 - - [01/Feb/2016 14:26:28] "GET /static/jquery.js HTTP/1.1" 304 0
192.168.1.246 - - [01/Feb/2016 14:26:28] "GET /static/foundation.min.js HTTP/1.1" 304 0
192.168.1.246 - - [01/Feb/2016 14:26:33] "POST /auth HTTP/1.1" 303 0
192.168.1.246 - - [01/Feb/2016 14:26:42] "GET /login/auth HTTP/1.1" 200 1079

Is there a way to get additional debug output?

**Comment by justinyhuang** [comment]
same issue with jbobspants here. using Raspbian on Raspberry Pi 2. Pianobar 2014.06.08.
what info could help you debug further?

thanks,

**Comment by tonymcgivern** [comment]
Same issue I'm afraid, Jessie latest, Pianobar 2014.06.08.

**Comment by jbobspants** [comment]
I dug my old Raspi out of storage recently and ran into this issue again trying to set up a headless music center, also duplicated it on an Odroid C1+ running Volumio 2. I dug around in the code and by adding some debugging output and conditionally extending the login iteration, I was able to get it working. I'm not sure why, but it seems that the output of pianobar's login sequence may vary between platforms.

Start at line 78 in pianobar_web.py (this is in the authenticate() function), comment out the below line:
`auth = [proc.stdout.readline() for i in range(0, 4)][-1]`

and replace it with:
```
#auth = [proc.stdout.readline() for i in range(0, 4)][-1]
for i in range(6):
    auth = proc.stdout.readline()
    print(str(i) + ": \n" + str(auth))
    if auth == b'\x1b[2K(i) Login... Ok.\n':
        print("detected login success")
        break;
```


Additionally, I had to make sure pianobar didn't try to automatically play the last station, but editting ~/.config/pianobar/state and removing the line about "last station" or "auto_start" or something...sorry can't remember exactly what it was. The only lines left in that file for me are volume=0 and the note about not editing the file.
